lib/GLPI/Agent/SNMP/MibSupport/BrotherNetConfig.pm

```diff
@@ -77,6 +81,34 @@ sub getModel {
     return $model;
 }
 
+sub getPortIfType {
+    my ($self) = @_;
+
+    my $device = $self->device
+        or return;
+
+    # Get list of device ports
+    my $ports = $device->{PORTS}->{PORT};
+
+    # Get list of device ports types (lan(1)/wirelesslan(2))
+    my $brMultiIFType = $self->walk(brMultiIFType) || {};
+
+    # Get list of device ports names
+    my $brMultiIFNodeType = $self->walk(brMultiIFNodeType) || {};
+
+    foreach my $index (keys %{$brMultiIFType}) {
+        foreach my $port (keys %{$ports}) {
+            if ($ports->{$port}->{IFNAME} eq $brMultiIFNodeType->{$index}) {
+                # wirelesslan(2)
+                if (defined($brMultiIFType->{$index}) && $brMultiIFType->{$index} eq 2) {
```

Comment:
You should better check it is an integer. And there's a function for that in GLPI::Agent::SNMP::Tools. Than a numeric test would be cleaner.
```suggestion
                # wirelesslan(2)
                if (defined($brMultiIFType->{$index}) && isInteger($brMultiIFType->{$index}) && int($brMultiIFType->{$index}) == 2) {
```